third_party/terraform/resources/resource_compute_instance_group_manager.go.erb

```diff
@@ -525,14 +535,21 @@ func resourceComputeInstanceGroupManagerRead(d *schema.ResourceData, meta interf
 		update_strategy = "REPLACE"
 	}
 	d.Set("update_strategy", update_strategy.(string))
+
+	// Terraform _really_ doesn't like removing lists.
```

Comment:
What does this mean? Why not? What happens?

Reply:
Ah I ran into like 5 occurrences of this in the same day and was not very patient with it. I'll update the comment to be better.

We get a permadiffdiff on the `.#` of a list from `0 => ` (empty-as-in-nil).

Reply:
Would a state migration fix this? I think if we just run a state migration to remove that field entirely, that's.... safer in the long run, I believe?

Reply:
I don't think a state migration would work because this happens on new resources. The GKE CI tests failing the last few days were the same issue, fixed by https://github.com/GoogleCloudPlatform/magic-modules/pull/922

Reply:
That's... concerning. If a field is marked as removed, it shouldn't wind up in state, I don't think? Let me verify that understanding really quick.

Reply:
Just to verify: this permadiff happened when there were no `d.Set` calls setting the field, right? The `d.Set` was removed, the field was marked Removed, and then the tests were run, and they resulted in a permadiff?

Reply:
Yes!

Reply:
Very bizarre. Let me see if I can dig up some information.